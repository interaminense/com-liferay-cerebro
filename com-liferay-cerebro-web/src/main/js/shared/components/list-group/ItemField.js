import Component, {Config} from 'metal-jsx';
import getCN from 'classnames';

class ItemField extends Component {
	render() {
		const {children, expand} = this.props;

		const classes = getCN('flex-col', {
			'flex-col-expand': expand
		});

		return (
			<div {...this.otherProps()} class={classes}>
				{children}
			</div>
		);
	}
}

ItemField.PROPS = {
	expand: Config.bool().value(false)
};

export default ItemField;