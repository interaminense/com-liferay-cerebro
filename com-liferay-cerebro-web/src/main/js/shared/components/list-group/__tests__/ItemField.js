import ItemField from '../ItemField';

describe('ItemField', () => {
	let component;

	afterEach(() => {
		if (component) {
			component.dispose();
		}
	});

	it('should render', () => {
		component = new ItemField();

		expect(component).toMatchSnapshot();
	});
});