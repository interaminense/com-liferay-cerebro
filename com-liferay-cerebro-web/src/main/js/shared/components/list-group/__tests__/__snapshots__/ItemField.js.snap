// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ItemField should render 1`] = `<div class="flex-col"></div>`;
